### PR TITLE
fix: headers and cors

### PR DESCRIPTION
**Pull request by markusahlstrand**
A few things that needed to be updated:
- Patched and pushed a new version of the tsoa-hono library to set response headers correctly
- CORS-config.. took a while to get this right
- Added the node-polyfill as we're using node buffers.. Think this isn't ideal, but it works

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,15 +49,15 @@
     "bcryptjs": "^2.4.3",
     "body-parser": "^1.20.2",
     "cookie": "^0.5.0",
-    "hono": "^3.6.3",
+    "hono": "^3.9.0",
```
Comment (markusahlstrand) [inline package.json:52]: Updated to fix headers and some other patches.

File: package.json [hunk 2]
```diff
@@ -49,15 +49,15 @@
...
     "kysely": "^0.26.3",
     "kysely-d1": "^0.3.0",
     "kysely-planetscale": "^1.4.0",
     "liquidjs": "^10.9.2",
     "mime": "^3.0.0",
     "trpc-durable-objects": "^1.3.4",
     "tsoa": "^5.1.1",
-    "tsoa-hono": "^1.0.3",
+    "tsoa-hono": "^1.0.9",
```
Comment (markusahlstrand) [inline package.json:60]: Updated to get the response header patches and few other updates

File: src/authentication.ts [hunk 3]
```diff
@@ -214,7 +214,7 @@ export async function verifyTenantPermissions(
   // Check db permissions
   const { members } = await ctx.env.data.members.list(tenantId);
   const member = members.find(
-    (m) => m.sub === ctx.state.user.sub && m.status === "active",
+    (m) => m.sub === ctx.var.user.sub && m.status === "active",
```
Comment (markusahlstrand) [inline src/authentication.ts:217]: State doesn't exist now.. It's ctx.var instead.

File: wrangler.toml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 name = "auth2"
 main = "src/server.ts"
 compatibility_date = "2022-07-06"
+node_compat = true
```
Comment (markusahlstrand) [inline wrangler.toml:4]: This is apparently not great.. We only need buffers and even that is probably something we can work around.

**Comment by markusahlstrand** [comment]
As we release the hono stuff to dev this is kind of blocking the release pipeline.. I'll go ahead and merge this one, but please still review it :)

File: package.json (hunk 2 above)
Comment (danielgent) [reply package.json:60]: Should someone be reviewing this library? :smile: 

Great way to sneak in unreviewed code into a PR, put it in your own OSS library :sunglasses:  I haven't seen that before

File: src/app.ts [hunk 5]
```diff
@@ -23,8 +23,26 @@ app.onError((err, ctx) => {
   return ctx.text(err.message, 500);
 });
 
+app.use(
+  "*",
+  cors({
+    origin: "*",
```
Comment (danielgent) [inline src/app.ts:29]: :eyes: this is what you were mentioning?

**Pull request merged**